forc-pkg/src/pkg.rs

```diff
@@ -2110,9 +2108,9 @@ pub fn check(
 
         let ast_result = sway_core::parsed_to_ast(parse_program, dep_namespace);
 
-        let typed_program = match &ast_result {
-            CompileAstResult::Failure { .. } => bail!("unable to type check"),
-            CompileAstResult::Success { typed_program, .. } => typed_program,
+        let typed_program = match &ast_result.value {
+            None => bail!("unable to type check"),
```

Comment:
We need to return the warnings and errors here so was can access them in LSP. 

Same above where we match on `parsed_result.value`. Errors and warnings need to be returned.